src/plugins/discover/public/application/main/hooks/utils/build_state_subscribe.ts

```diff
@@ -57,8 +57,8 @@ export const buildStateSubscribe =
         return;
       }
       savedSearch.searchSource.setField('index', nextDataView);
-      stateContainer.actions.setDataView(nextDataView);
       stateContainer.dataState.reset();
+      stateContainer.actions.setDataView(nextDataView);
```

Comment:
@davismcphee good performance catch! I've exchanged the order of resetting the data state and the setting of the next data view. and so the performance is like before, it was one of the changes I've suggested to ease the pain of https://github.com/elastic/kibana/pull/149294 , and it seems the re-rendering it triggered caused the performance issue (when you had just a single row, it didn't matter, but with many rows, et voila!, bad)

Reply:
FYI @mattkime one of the changes that need to be reverted in https://github.com/elastic/kibana/pull/149294, but better once this is merged, and all testing is done